0000-event-system.md

```diff
@@ -0,0 +1,142 @@
+* Start Date: 2018-01-16
+* RFC PR: (leave this empty)
+* React Navigation Issue: (leave this empty)
+
+# Summary
+
+Add an event emitter to the navigation prop that emits focus/blur events for the route.
+
+This proposal has been pioneered by @satya164: https://github.com/react-navigation/react-navigation/issues/1363 https://github.com/react-navigation/react-navigation/issues/51
+
+# Basic example
+
+In this example, we focus the first text input when the screen has completed transitioning:
+
+```
+componentDidMount() {
+  this._sub = this.props.navigation.addEventListener(
+    'onDidFocus',
```

Comment:
I think it'll be better name these events as just `didFocus`, `didBlur` etc. without the `on` prefix:

`onDidFocus` -> a prop to a react component, or dom element etc., similar to `onClick`, or a property on a element instance e.g. - `socket.onopen = () => { ... }`
`didFocus` -> similar to `addEventListener('click', () => { ... })` etc.

Reply:
@satya164 - naming was inspired by https://reactjs.org/docs/events.html#reference

Reply:
@brentvatne those names are name of the props to components, e.g. `<button onClick>` and similar to `<button onclick>` in HTML, I think these names don't fit well with an `addEventListener` API which drops the `on` prefix